x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/action/TransportDeleteEnrichPolicyAction.java

```diff
@@ -69,9 +80,12 @@ protected AcknowledgedResponse read(StreamInput in) throws IOException {
     @Override
     protected void masterOperation(Task task, DeleteEnrichPolicyAction.Request request, ClusterState state,
                                    ActionListener<AcknowledgedResponse> listener) throws Exception {
+        EnrichPolicy policy = EnrichStore.getPolicy(request.getName(), state); // ensure the policy exists first
+        if (policy == null) {
+            throw new ResourceNotFoundException("policy [{}] not found", request.getName());
+        }
         enrichPolicyLocks.lockPolicy(request.getName());
```

Comment:
It would be quite simple to do in this PR. I already had to add release logic to this PR, so i can just move it and remove a bit of code in the process.